hashicorp/dev-portal
# refactor: central docs URL adjusts, use MdxA

## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

- Consolidated docs URL adjustment to server-side functions
- Adds an `MdxA` component for link rendering in MDX content
- Replaces `DocsAnchor` with `MdxA`

## 🤷 Why

- To ensure consistency in link styling in MDX
- To make it easier to reason about URL adjustments in docs content, by centralizing logic

## 🛠️ How

- Refactors server-side `productUrlAdjusters` to do work previously done client-side by `DocsAnchor`
- Uses link adjusting remark plugin with `productUrlAdjusters` to docs view server code (in `layout/sidebar-sidecar`)

## 🧪 Testing

- Visit docs pages for beta products, links should appear and function as expected, with correct URLs
    - [vault][/vault/docs] - ⭐ focus area for review, since already in Beta
    - [waypoint][/waypoint/docs] - ⭐ focus area for review, since already in Beta
- Visit docs pages for all other products, links should appear and function as expected, with correct URLs
    - [boundary][/boundary/docs]
    - [consul][/consul/docs]
    - [hcp][/hcp/docs]
    - [nomad][/nomad/docs]
    - [packer][/packer/docs]
    - [sentinel][/sentinel/docs]
    - [terraform][/terraform/docs]
    - [vagrant][/vagrant/docs]


## 💭 Anything else?

Not at the moment.

[task]: https://app.asana.com/0/1202097197789424/1202569127502494/f
[preview]: https://dev-portal-git-zsshared-mdx-anchor-hashicorp.vercel.app
[/boundary/docs]: https://dev-portal-git-zsshared-mdx-anchor-hashicorp.vercel.app/boundary/docs
[/consul/docs]: https://dev-portal-git-zsshared-mdx-anchor-hashicorp.vercel.app/consul/docs
[/hcp/docs]: https://dev-portal-git-zsshared-mdx-anchor-hashicorp.vercel.app/hcp/docs
[/nomad/docs]: https://dev-portal-git-zsshared-mdx-anchor-hashicorp.vercel.app/nomad/docs
[/packer/docs]: https://dev-portal-git-zsshared-mdx-anchor-hashicorp.vercel.app/packer/docs
[/sentinel/docs]: https://dev-portal-git-zsshared-mdx-anchor-hashicorp.vercel.app/sentinel/docs
[/terraform/docs]: https://dev-portal-git-zsshared-mdx-anchor-hashicorp.vercel.app/terraform/docs
[/vagrant/docs]: https://dev-portal-git-zsshared-mdx-anchor-hashicorp.vercel.app/vagrant/docs
[/vault/docs]: https://dev-portal-git-zsshared-mdx-anchor-hashicorp.vercel.app/vault/docs
[/waypoint/docs]: https://dev-portal-git-zsshared-mdx-anchor-hashicorp.vercel.app/waypoint/docs
